### PR TITLE
MOL-224/Behat tests for reccuring payments

### PR DESCRIPTION
**Pull request by marekrzytki**

### CODE REVIEW

File: tests/Behat/Context/Setup/MollieContext.php [hunk 1]
```diff
@@ -35,16 +41,40 @@ final class MollieContext implements Context
...
 
+    private string $mollieApiKeyTest;
+
+    private string $mollieProfileId;
+
+    private MollieMethodsResolverInterface $mollieMethodsResolver;
+
+    private MolliePaymentMethodPurifierInterface $molliePaymentMethodPurifier;
+
+    private EntityRepository $gatewayConfigRepository;
```
Comment (senghe) [inline tests/Behat/Context/Setup/MollieContext.php:52]: RepositoryInterface or dedicated repository interface?

File: tests/Behat/Context/Setup/MollieContext.php [hunk 2]
```diff
@@ -62,11 +92,34 @@ public function theStoreHasAPaymentMethodWithACodeAndMolliePaymentGateway(string
...
+    public function gatewayHasAllMethodsLoadedAndEnabled(string $paymentMethodCode): void
+    {
+        /** @var GatewayConfigInterface $gatewayConfig */
+        $gatewayConfig = $this->gatewayConfigRepository
+            ->findOneBy([
+                'gatewayName' => $paymentMethodCode,
+            ]);
+
+        if (null === $gatewayConfig) {
+            throw new ResourceNotFoundException('Gateway not found');
```
Comment (senghe) [inline tests/Behat/Context/Setup/MollieContext.php:115]: How about putting gateway name into exception message?

File: tests/Behat/Context/Setup/MollieContext.php [hunk 3]
```diff
@@ -82,13 +135,32 @@ public function theStoreHasAPaymentMethodWithACodeAndMollieSubscriptionPaymentGa
...
+    }
+
+    private function enableAllMolliePaymentMethods(): void
+    {
+        $molliePaymentMethods = $this->mollieConfigurationRepository->findAll();
+
+        /** @var MollieGatewayConfigInterface $molliePaymentMethod */
+        foreach ($molliePaymentMethods as $molliePaymentMethod) {
+            $molliePaymentMethod->enable();
+            $molliePaymentMethod->setCountryRestriction(MollieGatewayConfigInterface::ALL_COUNTRIES);
```
Comment (senghe) [inline tests/Behat/Context/Setup/MollieContext.php:160]: How about splitting the long line into smaller ones?

File: tests/Behat/Context/Ui/Admin/ManagingPaymentMethodContext.php [hunk 4]
```diff
@@ -39,6 +48,12 @@ public function iWantToCreateANewMolliePaymentMethod(): void
      */
     public function iConfigureItWithTestMollieCredentials(string $apiKey): void
     {
+        if ('MOLLIE_TEST_API_KEY' === $apiKey) {
```
Comment (senghe) [inline tests/Behat/Context/Ui/Admin/ManagingPaymentMethodContext.php:51]: Could you put it into constant?

File: tests/Behat/Context/Ui/Admin/ManagingPaymentMethodContext.php [hunk 5]
```diff
@@ -47,6 +62,12 @@ public function iConfigureItWithTestMollieCredentials(string $apiKey): void
      */
     public function iConfigureProfileId(string $profileId): void
     {
+        if ('MOLLIE_PROFILE_KEY' === $profileId) {
```
Comment (senghe) [inline tests/Behat/Context/Ui/Admin/ManagingPaymentMethodContext.php:65]: Could you put it into constant?

File: tests/Behat/Page/Admin/PaymentMethod/CreatePage.php [hunk 6]
```diff
@@ -49,6 +49,18 @@ public function setInterval(string $interval): void
         $this->getDocument()->fillField('Interval', $interval);
     }
 
+    public function loadPaymentMethods(): void
+    {
+        $this->getDocument()->find('css', '#get_methods')->click();
+        $time = 25000;
```
Comment (senghe) [inline tests/Behat/Page/Admin/PaymentMethod/CreatePage.php:55]: How about checking where is the boundary for the value and choose the lower one.

**Review by senghe — CHANGES_REQUESTED** [review]

File: tests/Behat/Context/Setup/MollieContext.php (hunk 1 above)
Comment (marekrzytki) [reply tests/Behat/Context/Setup/MollieContext.php]: Done

File: tests/Behat/Context/Setup/MollieContext.php (hunk 2 above)
Comment (marekrzytki) [reply tests/Behat/Context/Setup/MollieContext.php]: Done

File: tests/Behat/Context/Setup/MollieContext.php (hunk 3 above)
Comment (marekrzytki) [reply tests/Behat/Context/Setup/MollieContext.php]: Done

File: tests/Behat/Context/Ui/Admin/ManagingPaymentMethodContext.php (hunk 4 above)
Comment (marekrzytki) [reply tests/Behat/Context/Ui/Admin/ManagingPaymentMethodContext.php]: Yes, sure

File: tests/Behat/Context/Ui/Admin/ManagingPaymentMethodContext.php (hunk 5 above)
Comment (marekrzytki) [reply tests/Behat/Context/Ui/Admin/ManagingPaymentMethodContext.php]: Yes, sure

File: tests/Behat/Page/Admin/PaymentMethod/CreatePage.php (hunk 6 above)
Comment (marekrzytki) [reply tests/Behat/Page/Admin/PaymentMethod/CreatePage.php]: I've reduced it and also add check for earlier execution

**Review by senghe — APPROVED** [review]

**Pull request merged**